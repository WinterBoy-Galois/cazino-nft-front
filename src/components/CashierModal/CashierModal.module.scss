@import '../../styles/shared';

.slide-select {
  max-width: 428px;
  margin: 0 auto;
  margin-bottom: 3rem;
}

.slide-select-withdraw {
  margin-bottom: 2.5rem;

  @include media-breakpoint-up(md) {
    margin-bottom: 1rem;
  }
}

.spacing {
  margin-bottom: 1rem;
}

.link {
  font-size: 18px;
  font-weight: 400;

  &__container {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
}

.qrcode {
  &__container {
    width: 200px;
    padding: 1rem;
    background-color: $white;
    border-radius: 8px;
    margin: 0 auto;
  }
}

.icon {
  width: 0.9rem;
  fill: $bitcoinYellow;
  margin-right: 0.5rem;

  &__inner {
    fill: $light-blue;
  }
}

.withdraw {
  &__container {
    background-color: #14233a;
    margin: 0;
    border-radius: 8px;
    margin-bottom: 1rem;
    padding: 15px;

    & > div {
      margin: 10px 0px;

      &:first-child {
        margin-top: 0;
      }

      &:last-child {
        margin-bottom: 0;
      }
    }
  }

  &__row {
    margin: auto;
  }

  &__label {
    align-items: center;
    display: flex;
    text-transform: uppercase;
    font-size: 16px;
    line-height: 21px;
    color: #7ea3cc;
    user-select: none;
    margin: 0;
  }

  &__value {
    display: flex;
    align-items: center;
    justify-content: flex-end;
  }

  &__amount_control {
    margin-bottom: 3rem;
  }

  &__button {
    width: 16rem;
    display: block;
    margin: auto;
    background-color: #f19e65;
    color: #4d290f;

    &:hover {
      background-color: #f3b489;
    }

    @include media-breakpoint-up(md) {
      width: 16.67rem;
    }
  }
}
