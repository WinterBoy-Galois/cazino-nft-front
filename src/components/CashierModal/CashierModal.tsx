import React, { Fragment, useCallback, useEffect, useState } from 'react';
import styles from './CashierModal.module.scss';
import Modal from '../Modal';
import SlideSelect from '../SlideSelect';
import { useStateValue } from '../../state';
import { useNavigate, useLocation } from '@reach/router';
import { useMutation, useQuery } from '@apollo/client';
import { SETUP_CASHIER } from '../../graphql/queries';
import Cashier from '../../models/cashier.model';
import CopyField from '../CopyField';
import Link from '../Link';
import Loading from '../Loading';
import Error from '../Error';
import QRCode from 'qrcode.react';
import TextInput from '../TextInput';
import { useTranslation } from 'react-i18next';
import Bitcoin from '../icons/social/Bitcoin';
import { formatBitcoin } from '../../common/util/format.util';
import Button from '../../components/Button';
import WithdrawAmountControl from '../../components/WithdrawAmountControl';
import { WITHDRAW } from '../../graphql/mutations';
import { success, error as errorToast } from '../../components/Toast';
import clsx from 'clsx';
import validate from 'bitcoin-address-validation';

interface IProps {
  show: boolean;
  loading: boolean;
  error?: any;
  onClose?: () => void;
  cashier?: Cashier;
  balance?: number;
  depositAddress?: string;
  onTransactionsLinkClick?: () => void;
}

const CashierModal: React.FC<IProps> = ({
  show,
  onClose,
  cashier,
  balance,
  loading,
  error,
  depositAddress: defaultDepositAddress = '',
  onTransactionsLinkClick,
}) => {
  const { t } = useTranslation(['modals']);
  const [modalType, setModalType] = useState('deposit');
  const [amount, setAmount] = useState(0);
  const [withdraw] = useMutation(WITHDRAW);
  const [isValidated, setValidated] = useState(false);
  const [depositAddress, setDepositAddress] = useState(defaultDepositAddress);

  useEffect(() => {
    if (validate(depositAddress)) setValidated(true);
    else setValidated(false);
  }, [depositAddress]);

  useEffect(() => setDepositAddress(defaultDepositAddress), [defaultDepositAddress]);

  return (
    <Modal show={show} onClose={onClose} title={t('cashier.title')}>
      {loading && <Loading />}
      {!loading && error && <Error />}
      {!loading && !error && !depositAddress && depositAddress !== '' && (
        <Error>{t('cashier.depositNullError')}</Error>
      )}

      {!loading && !error && cashier && (depositAddress || depositAddress === '') && (
        <Fragment>
          <SlideSelect
            className={clsx(
              styles['slide-select'],
              modalType === 'withdraw' ? styles['slide-select-withdraw'] : null
            )}
            selectItems={[
              { label: 'deposit', onClick: () => setModalType('deposit') },
              { label: 'withdraw', onClick: () => setModalType('withdraw') },
            ]}
          />

          {modalType === 'deposit' ? (
            <div className="row">
              <div className="col-12 col-md-8">
                <CopyField
                  label={t('cashier.bitcoinWalletAddress')}
                  value={depositAddress}
                  className={styles.spacing}
                />

                <TextInput
                  label={t('cashier.requiredConfirmations')}
                  value={`${cashier.depositConfirmations}`}
                  disabled
                />

                <div className={styles.link__container}>
                  <Link className={styles.link} onClick={onTransactionsLinkClick}>
                    {t('cashier.transactionsLinkText')}
                  </Link>
                </div>
              </div>
              <div className="col-12 col-md-4">
                <div className={styles.qrcode__container}>
                  <QRCode value={depositAddress} renderAs="svg" width={'100%'} height={'100%'} />
                </div>
              </div>
            </div>
          ) : modalType === 'withdraw' ? (
            <div className="row">
              <div className={clsx(styles.withdraw__row, 'col-12 col-md-10 col-lg-8')}>
                <div className={clsx(styles.withdraw__container, 'row')}>
                  <div className="col-12">
                    <div className="row">
                      <div className={clsx(styles.withdraw__label, 'col-6 col-md-7')}>Balance</div>
                      <div className={clsx(styles.withdraw__value, 'col-6 col-md-5')}>
                        <Bitcoin className={styles.icon} innerClassName={styles.icon__inner} />
                        <span>{formatBitcoin(balance)}</span>
                      </div>
                    </div>
                  </div>

                  <div className="col-12">
                    <div className="row">
                      <div className={clsx(styles.withdraw__label, 'col-6 col-md-7')}>
                        Blockchain Fee
                      </div>
                      <div className={clsx(styles.withdraw__value, 'col-6 col-md-5')}>
                        <Bitcoin className={styles.icon} innerClassName={styles.icon__inner} />
                        <span>{formatBitcoin(cashier.networkFee)}</span>
                      </div>
                    </div>
                  </div>

                  <div className="col-12">
                    <div className="row">
                      <div className={clsx(styles.withdraw__label, 'col-6 col-md-7')}>
                        Minimum Withdraw Amount
                      </div>
                      <div className={clsx(styles.withdraw__value, 'col-6 col-md-5')}>
                        <Bitcoin className={styles.icon} innerClassName={styles.icon__inner} />
                        <span>{formatBitcoin(cashier.minWithdraw)}</span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>

              <div className={clsx(styles.withdraw__row, 'col-12 col-md-10 col-lg-8')}>
                <TextInput
                  label={t('cashier.bitcoinWalletAddress')}
                  value={depositAddress}
                  validationMessage={isValidated ? '' : 'Address is not valid.'}
                  onChangeValue={depositAddress => setDepositAddress(depositAddress)}
                />
              </div>

              <div
                className={clsx(
                  styles.withdraw__row,
                  styles.withdraw__amount_control,
                  'col-12 col-md-10 col-lg-8'
                )}
              >
                <WithdrawAmountControl
                  amount={0}
                  max={balance ? balance - cashier.networkFee : 0}
                  onChange={amount => setAmount(amount)}
                />
              </div>

              <div className={clsx(styles.withdraw__row, 'col-12 col-md-10 col-lg-8')}>
                <Button
                  className={styles.withdraw__button}
                  onClick={async () => {
                    const { data, errors } = await withdraw({
                      variables: { amount: amount, address: depositAddress },
                    });

                    if (errors) errorToast('Withdraw is failed.');
                    else if (data?.withdraw?.result) success('Withdraw is completed.');
                    else errorToast('Withdraw is failed.');
                  }}
                >
                  Withdraw
                </Button>
              </div>
            </div>
          ) : null}
        </Fragment>
      )}
    </Modal>
  );
};

export default CashierModal;

export const CashierModalWithData: React.FC<IProps> = props => {
  const [{ auth }] = useStateValue();
  const navigate = useNavigate();
  const { pathname } = useLocation();
  const { data, loading, error, refetch } = useQuery(SETUP_CASHIER);
  const handleTransactionsClick = useCallback(() => navigate('/transactions/deposits'), [navigate]);

  useEffect(() => {
    if (!data?.me.depositAddress && !loading && props.show) {
      refetch();
    }
  }, [data, refetch, props.show, loading]);

  if (props.show && auth.state !== 'SIGNED_IN') {
    navigate(pathname);
    return null;
  }

  return (
    <CashierModal
      {...props}
      loading={loading}
      error={error}
      cashier={data?.setupCashier}
      balance={auth.user?.balance}
      depositAddress={data?.me.depositAddress}
      onTransactionsLinkClick={handleTransactionsClick}
    />
  );
};
