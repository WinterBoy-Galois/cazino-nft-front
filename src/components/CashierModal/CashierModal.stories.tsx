import React from 'react';
import { boolean, number, text } from '@storybook/addon-knobs';

import CashierModal from '.';

export default {
  title: 'Components/CashierModal',
  component: CashierModal,
};

const data = {
  show: true,
  cashier: {
    networkFee: 0.00075295,
    depositConfirmations: 2,
    minWithdraw: 2e-7,
  },
  depositAddress: '3CaQi76neM8qoDh18pLq5W8Hru1sNVRoDY',
};

export const Default = () => (
  <CashierModal
    show={boolean('Show', data.show)}
    loading={boolean('Loading', false)}
    error={boolean('Error', false)}
    cashier={{
      networkFee: number('Network fee', data.cashier.networkFee),
      depositConfirmations: number('Confirmations', data.cashier.depositConfirmations),
      minWithdraw: number('Min Withdraw', data.cashier.minWithdraw),
    }}
    depositAddress={text('Deposit Address', data.depositAddress)}
  />
);
