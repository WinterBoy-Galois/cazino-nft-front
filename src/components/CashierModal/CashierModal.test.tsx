import React from 'react';
import { render } from '@testing-library/react';
import CashierModal from '.';

describe('CashierModal', () => {
  xit('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(
      <CashierModal
        show
        loading={false}
        cashier={{
          networkFee: 0.00075295,
          depositConfirmations: 2,
          minWithdraw: 2e-7,
        }}
        depositAddress={'3CaQi76neM8qoDh18pLq5W8Hru1sNVRoDY'}
      />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
