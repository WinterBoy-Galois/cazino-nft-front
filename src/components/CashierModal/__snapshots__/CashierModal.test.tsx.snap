// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CashierModal should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="container"
        style="height: 768px;"
      >
        <div
          aria-modal="true"
          class="modal scale--enter scale--enter-active"
          role="dialog"
        >
          <div
            class="modal__header"
          >
            <div
              class="modal__header__headline"
            >
              Cashier
            </div>
            <div
              class="modal__header__close"
            >
              <svg
                class="icon icon__close"
                viewBox="0 0 365.696 365.696"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
                />
              </svg>
            </div>
          </div>
          <div
            class="modal__main"
          >
            <div
              class="container slide-select"
            >
              <span
                class="tab tab--selected"
                style="transform: translateX(0); width: 50%;"
              />
              <span
                class="tab"
                style="width: 50%;"
              >
                deposit
              </span>
              <span
                class="tab"
                style="width: 50%;"
              >
                withdraw
              </span>
            </div>
            <div
              class="row"
            >
              <div
                class="col-12 col-md-8"
              >
                <div
                  class="spacing container"
                >
                  <div
                    class="label"
                  >
                    Bitcoin Wallet Address
                  </div>
                  <div
                    class="value"
                  >
                    3CaQi76neM8qoDh18pLq5W8Hru1sNVRoDY
                  </div>
                  <div
                    class="copy"
                  >
                    <svg
                      class="default"
                      viewBox="0 0 12.14 13.88"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        data-name="Layer 2"
                      >
                        <g
                          data-name="Layer 1"
                        >
                          <path
                            d="M8.67 12.14v1.09a.66.66 0 01-.65.65H.65a.66.66 0 01-.65-.65v-10a.65.65 0 01.65-.63h2v8a1.52 1.52 0 001.52 1.52zm0-9.32V0H4.12a.65.65 0 00-.65.65v10a.65.65 0 00.65.65h7.37a.65.65 0 00.65-.65V3.47H9.32a.65.65 0 01-.65-.65zM12 2L10.16.19A.66.66 0 009.7 0h-.16v2.6h2.6v-.16A.66.66 0 0012 2z"
                            data-name="Icon awesome-copy"
                          />
                        </g>
                      </g>
                    </svg>
                  </div>
                </div>
                <div
                  class="inputField__container"
                >
                  <div
                    class="inputField__wrapper inputField__wrapper--disabled"
                  >
                    <label
                      class="inputField__label"
                    >
                      Required Confirmation(s)
                    </label>
                    <input
                      autocomplete="off"
                      class="inputField"
                      disabled=""
                      type="text"
                      value="2"
                    />
                  </div>
                </div>
                <div
                  class="link__container"
                >
                  <a
                    class="link link"
                    href=""
                  >
                    Deposit Transactions
                  </a>
                </div>
              </div>
              <div
                class="col-12 col-md-4"
              >
                <div
                  class="qrcode__container"
                >
                  <svg
                    height="100%"
                    shape-rendering="crispEdges"
                    viewBox="0 0 29 29"
                    width="100%"
                  >
                    <path
                      d="M0,0 h29v29H0z"
                      fill="#FFFFFF"
                    />
                    <path
                      d="M0 0h7v1H0zM8 0h1v1H8zM11 0h2v1H11zM15 0h2v1H15zM20 0h1v1H20zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM8 1h3v1H8zM12 1h3v1H12zM16 1h1v1H16zM20 1h1v1H20zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM10 2h3v1H10zM14 2h5v1H14zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM10 3h1v1H10zM14 3h1v1H14zM18 3h1v1H18zM20 3h1v1H20zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h4v1H8zM13 4h3v1H13zM18 4h2v1H18zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM11 5h1v1H11zM15 5h1v1H15zM20 5h1v1H20zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM8 7h1v1H8zM10 7h3v1H10zM14 7h3v1H14zM20 7h1v1H20zM0 8h3v1H0zM5 8h2v1H5zM8 8h3v1H8zM13 8h2v1H13zM16 8h1v1H16zM19 8h1v1H19zM21 8h4v1H21zM27,8 h2v1H27zM0 9h1v1H0zM5 9h1v1H5zM7 9h1v1H7zM9 9h2v1H9zM13 9h2v1H13zM18 9h2v1H18zM22 9h2v1H22zM27 9h1v1H27zM0 10h1v1H0zM2 10h2v1H2zM5 10h2v1H5zM8 10h1v1H8zM11 10h1v1H11zM15 10h1v1H15zM18 10h2v1H18zM24 10h1v1H24zM26 10h1v1H26zM28,10 h1v1H28zM0 11h1v1H0zM5 11h1v1H5zM9 11h1v1H9zM13 11h1v1H13zM21 11h2v1H21zM25 11h1v1H25zM27 11h1v1H27zM1 12h6v1H1zM9 12h1v1H9zM11 12h3v1H11zM19 12h1v1H19zM22 12h1v1H22zM25 12h1v1H25zM27,12 h2v1H27zM0 13h1v1H0zM2 13h3v1H2zM8 13h2v1H8zM12 13h1v1H12zM16 13h1v1H16zM18 13h3v1H18zM23 13h1v1H23zM26 13h1v1H26zM28,13 h1v1H28zM3 14h4v1H3zM8 14h3v1H8zM12 14h6v1H12zM19 14h2v1H19zM22 14h3v1H22zM26 14h1v1H26zM28,14 h1v1H28zM1 15h2v1H1zM10 15h3v1H10zM14 15h1v1H14zM16 15h1v1H16zM18 15h1v1H18zM20 15h1v1H20zM24 15h1v1H24zM27,15 h2v1H27zM1 16h1v1H1zM3 16h2v1H3zM6 16h2v1H6zM13 16h4v1H13zM18 16h1v1H18zM20 16h1v1H20zM22 16h1v1H22zM27,16 h2v1H27zM1 17h5v1H1zM10 17h1v1H10zM13 17h2v1H13zM16 17h1v1H16zM18 17h1v1H18zM20 17h1v1H20zM22 17h1v1H22zM25 17h1v1H25zM0 18h4v1H0zM6 18h2v1H6zM9 18h1v1H9zM11 18h1v1H11zM15 18h1v1H15zM17 18h5v1H17zM24 18h1v1H24zM28,18 h1v1H28zM7 19h2v1H7zM10 19h1v1H10zM13 19h1v1H13zM15 19h2v1H15zM18 19h1v1H18zM23 19h2v1H23zM27,19 h2v1H27zM0 20h3v1H0zM4 20h1v1H4zM6 20h1v1H6zM9 20h1v1H9zM11 20h3v1H11zM16 20h1v1H16zM19 20h6v1H19zM27,20 h2v1H27zM8 21h2v1H8zM12 21h1v1H12zM15 21h2v1H15zM18 21h3v1H18zM24 21h2v1H24zM27,21 h2v1H27zM0 22h7v1H0zM12 22h5v1H12zM18 22h1v1H18zM20 22h1v1H20zM22 22h1v1H22zM24 22h2v1H24zM27,22 h2v1H27zM0 23h1v1H0zM6 23h1v1H6zM8 23h2v1H8zM11 23h2v1H11zM14 23h2v1H14zM18 23h3v1H18zM24 23h2v1H24zM28,23 h1v1H28zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM10 24h1v1H10zM13 24h3v1H13zM17 24h1v1H17zM19 24h7v1H19zM27,24 h2v1H27zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM10 25h1v1H10zM13 25h2v1H13zM16 25h2v1H16zM25 25h2v1H25zM28,25 h1v1H28zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM8 26h4v1H8zM15 26h2v1H15zM19 26h1v1H19zM24 26h2v1H24zM27,26 h2v1H27zM0 27h1v1H0zM6 27h1v1H6zM8 27h3v1H8zM13 27h1v1H13zM15 27h2v1H15zM19 27h2v1H19zM23 27h1v1H23zM25 27h1v1H25zM0 28h7v1H0zM8 28h6v1H8zM15 28h1v1H15zM17 28h1v1H17zM19 28h2v1H19zM22 28h1v1H22zM24 28h2v1H24zM28,28 h1v1H28z"
                      fill="#000000"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="height: 768px;"
    >
      <div
        aria-modal="true"
        class="modal scale--enter scale--enter-active"
        role="dialog"
      >
        <div
          class="modal__header"
        >
          <div
            class="modal__header__headline"
          >
            Cashier
          </div>
          <div
            class="modal__header__close"
          >
            <svg
              class="icon icon__close"
              viewBox="0 0 365.696 365.696"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
              />
            </svg>
          </div>
        </div>
        <div
          class="modal__main"
        >
          <div
            class="container slide-select"
          >
            <span
              class="tab tab--selected"
              style="transform: translateX(0); width: 50%;"
            />
            <span
              class="tab"
              style="width: 50%;"
            >
              deposit
            </span>
            <span
              class="tab"
              style="width: 50%;"
            >
              withdraw
            </span>
          </div>
          <div
            class="row"
          >
            <div
              class="col-12 col-md-8"
            >
              <div
                class="spacing container"
              >
                <div
                  class="label"
                >
                  Bitcoin Wallet Address
                </div>
                <div
                  class="value"
                >
                  3CaQi76neM8qoDh18pLq5W8Hru1sNVRoDY
                </div>
                <div
                  class="copy"
                >
                  <svg
                    class="default"
                    viewBox="0 0 12.14 13.88"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      data-name="Layer 2"
                    >
                      <g
                        data-name="Layer 1"
                      >
                        <path
                          d="M8.67 12.14v1.09a.66.66 0 01-.65.65H.65a.66.66 0 01-.65-.65v-10a.65.65 0 01.65-.63h2v8a1.52 1.52 0 001.52 1.52zm0-9.32V0H4.12a.65.65 0 00-.65.65v10a.65.65 0 00.65.65h7.37a.65.65 0 00.65-.65V3.47H9.32a.65.65 0 01-.65-.65zM12 2L10.16.19A.66.66 0 009.7 0h-.16v2.6h2.6v-.16A.66.66 0 0012 2z"
                          data-name="Icon awesome-copy"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
              </div>
              <div
                class="inputField__container"
              >
                <div
                  class="inputField__wrapper inputField__wrapper--disabled"
                >
                  <label
                    class="inputField__label"
                  >
                    Required Confirmation(s)
                  </label>
                  <input
                    autocomplete="off"
                    class="inputField"
                    disabled=""
                    type="text"
                    value="2"
                  />
                </div>
              </div>
              <div
                class="link__container"
              >
                <a
                  class="link link"
                  href=""
                >
                  Deposit Transactions
                </a>
              </div>
            </div>
            <div
              class="col-12 col-md-4"
            >
              <div
                class="qrcode__container"
              >
                <svg
                  height="100%"
                  shape-rendering="crispEdges"
                  viewBox="0 0 29 29"
                  width="100%"
                >
                  <path
                    d="M0,0 h29v29H0z"
                    fill="#FFFFFF"
                  />
                  <path
                    d="M0 0h7v1H0zM8 0h1v1H8zM11 0h2v1H11zM15 0h2v1H15zM20 0h1v1H20zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM8 1h3v1H8zM12 1h3v1H12zM16 1h1v1H16zM20 1h1v1H20zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM10 2h3v1H10zM14 2h5v1H14zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM10 3h1v1H10zM14 3h1v1H14zM18 3h1v1H18zM20 3h1v1H20zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h4v1H8zM13 4h3v1H13zM18 4h2v1H18zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM11 5h1v1H11zM15 5h1v1H15zM20 5h1v1H20zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM8 7h1v1H8zM10 7h3v1H10zM14 7h3v1H14zM20 7h1v1H20zM0 8h3v1H0zM5 8h2v1H5zM8 8h3v1H8zM13 8h2v1H13zM16 8h1v1H16zM19 8h1v1H19zM21 8h4v1H21zM27,8 h2v1H27zM0 9h1v1H0zM5 9h1v1H5zM7 9h1v1H7zM9 9h2v1H9zM13 9h2v1H13zM18 9h2v1H18zM22 9h2v1H22zM27 9h1v1H27zM0 10h1v1H0zM2 10h2v1H2zM5 10h2v1H5zM8 10h1v1H8zM11 10h1v1H11zM15 10h1v1H15zM18 10h2v1H18zM24 10h1v1H24zM26 10h1v1H26zM28,10 h1v1H28zM0 11h1v1H0zM5 11h1v1H5zM9 11h1v1H9zM13 11h1v1H13zM21 11h2v1H21zM25 11h1v1H25zM27 11h1v1H27zM1 12h6v1H1zM9 12h1v1H9zM11 12h3v1H11zM19 12h1v1H19zM22 12h1v1H22zM25 12h1v1H25zM27,12 h2v1H27zM0 13h1v1H0zM2 13h3v1H2zM8 13h2v1H8zM12 13h1v1H12zM16 13h1v1H16zM18 13h3v1H18zM23 13h1v1H23zM26 13h1v1H26zM28,13 h1v1H28zM3 14h4v1H3zM8 14h3v1H8zM12 14h6v1H12zM19 14h2v1H19zM22 14h3v1H22zM26 14h1v1H26zM28,14 h1v1H28zM1 15h2v1H1zM10 15h3v1H10zM14 15h1v1H14zM16 15h1v1H16zM18 15h1v1H18zM20 15h1v1H20zM24 15h1v1H24zM27,15 h2v1H27zM1 16h1v1H1zM3 16h2v1H3zM6 16h2v1H6zM13 16h4v1H13zM18 16h1v1H18zM20 16h1v1H20zM22 16h1v1H22zM27,16 h2v1H27zM1 17h5v1H1zM10 17h1v1H10zM13 17h2v1H13zM16 17h1v1H16zM18 17h1v1H18zM20 17h1v1H20zM22 17h1v1H22zM25 17h1v1H25zM0 18h4v1H0zM6 18h2v1H6zM9 18h1v1H9zM11 18h1v1H11zM15 18h1v1H15zM17 18h5v1H17zM24 18h1v1H24zM28,18 h1v1H28zM7 19h2v1H7zM10 19h1v1H10zM13 19h1v1H13zM15 19h2v1H15zM18 19h1v1H18zM23 19h2v1H23zM27,19 h2v1H27zM0 20h3v1H0zM4 20h1v1H4zM6 20h1v1H6zM9 20h1v1H9zM11 20h3v1H11zM16 20h1v1H16zM19 20h6v1H19zM27,20 h2v1H27zM8 21h2v1H8zM12 21h1v1H12zM15 21h2v1H15zM18 21h3v1H18zM24 21h2v1H24zM27,21 h2v1H27zM0 22h7v1H0zM12 22h5v1H12zM18 22h1v1H18zM20 22h1v1H20zM22 22h1v1H22zM24 22h2v1H24zM27,22 h2v1H27zM0 23h1v1H0zM6 23h1v1H6zM8 23h2v1H8zM11 23h2v1H11zM14 23h2v1H14zM18 23h3v1H18zM24 23h2v1H24zM28,23 h1v1H28zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM10 24h1v1H10zM13 24h3v1H13zM17 24h1v1H17zM19 24h7v1H19zM27,24 h2v1H27zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM10 25h1v1H10zM13 25h2v1H13zM16 25h2v1H16zM25 25h2v1H25zM28,25 h1v1H28zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM8 26h4v1H8zM15 26h2v1H15zM19 26h1v1H19zM24 26h2v1H24zM27,26 h2v1H27zM0 27h1v1H0zM6 27h1v1H6zM8 27h3v1H8zM13 27h1v1H13zM15 27h2v1H15zM19 27h2v1H19zM23 27h1v1H23zM25 27h1v1H25zM0 28h7v1H0zM8 28h6v1H8zM15 28h1v1H15zM17 28h1v1H17zM19 28h2v1H19zM22 28h1v1H22zM24 28h2v1H24zM28,28 h1v1H28z"
                    fill="#000000"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
